app/models/mixins/relationship_mixin.rb

```diff
@@ -54,6 +55,22 @@ def clear_relationships_cache(*args)
     @association_cache.delete(:all_relationships)
   end
 
+  # this allows us to call similarly named ancestry or mixin methods
+  # depending on whether default_relationship_type is set to :genealogy or not
+
+  def call_ancestry_method(method_name, *args)
+    bound_method = ((@@ancestry_method ||= {})[method_name] ||= ::Ancestry::InstanceMethods.instance_method(method_name)).bind(self)
+    if !args.first.nil? && args.flatten != [] && args.flatten != [{}] # ? !args.empty?
```

Comment:
wondering if the `*args` change will let you simplify this back to code that you want

I really really wish this were only `bound_method.call(*args)`